docs/start/api.query.multi.md

```diff
@@ -0,0 +1,60 @@
+# Multi queries
+
+In a number of applications, it is useful to monitor a number of like-queries at the same time. For instance, we may want to track the balances for a list of accounts we have. The `api.query` interfaces allows this via the `.multi` sub-call.
+
+## Multi queries, same type
+
+For queries of the same type, for instance to retrieve the balances of a number of accounts at once, we can do -
+
+```js
+...
+
+// subscribe to balance changes for 2 accounts, ADDR1 & ADDR2 (already defined)
+const unsub = await api.query.balances.freeBalance.multi([ADDR1, ADDR2], (balances) => {
+  const [balance1, balance2] = balances;
+
+  console.log(`The balances are ${balance1} and ${balance2}`);
+});
+```
+
+A couple of items to note in the example above: we don't call `freeBalance` directly, but rather `freeBalance.multi`. We pass the addresses we want to query as an array, obviously this would depend on the number of addresses we want to query. As an example, we can track the balances of a list of validators,
+
+```js
+...
+
+// retrieve a snapshot of the validators
+const validators = await api.query.session.validators();
+
+// subscribe to the balances for these accounts
+const unsub = await api.query.balances.freeBalance.multi(validators, (balances) => {
+  console.log(`The balances are: ${balances}`);
+});
+```
+
+The above example does not subscribe to the validators explicitly, but only gets a snapshot and used this into the future. It should be trivially extendendable to  subscribe to the validators, track which one have entered or left and then subscribe to balances as they change through the next blocks.
+
+## Multi queries, distinct types
+
+The previous `.multi` examples assumes that we do queries for the same types, i.e. we retrieve the balances for a number of accounts. However, there is also a need to retrieve various distinct types, as an example we would like to track the block timestamp in addition to the nonce and balance of a specific account. To cater for this, the api hs a specific `api.queryMulti` interface that can be used to perform this query -
+
+```js
+...
+
+// subscribe to the timestamp, our index and balance
+const unsub = await api.queryMulti([
+  api.query.timestamp.now,
+  [api.query.system.accountNonce, ADDR],
+  [api.query.balances.freeBalance, ADDR]
+], ([now, nonce, balance]) => {
+  console.log(`${now}: balance of ${balance} and a nonce of ${nonce}`);
+});
+```
```

Comment:
What's the syntax for making a multi-query when one of the queries expects multiple parameters like in `api.query.session.nextKeys`? My guess was:
```javascript
const unsub = await api.queryMulti([
  api.query.timestamp.now,
  [api.query.session.nextKeys, BYTES, ADDR]
], ([now, keys]) => {
  ...
});
```
But experimentation suggests it is
```javascript
const unsub = await api.queryMulti([
  api.query.timestamp.now,
  [api.query.session.nextKeys, [BYTES, ADDR]]
], ([now, keys]) => {
  ...
});
```

Reply:
I like the way you explain a similar concept to `.at`:
> In the above example, we introduce the `.at(<hash>, [params])` query.

Reply:
FWIW, I would have guessed correctly on the first try if the example had been
```javascript
  const unsub = await api.queryMulti([
    [api.query.timestamp.now, []],
    [api.query.system.accountNonce, [ADDR]],
    [api.query.balances.freeBalance, [ADDR]]
  ], ([now, nonce, balance]) => {
    console.log(`${now}: balance of ${balance} and a nonce of ${nonce}`);
  });
```

Reply:
`nextKeys` is one of the "funny" ones. Since it is a `DoubleMap`, the `[bytes, key]` needs to be a single variable, i.e. they are combined into 1 to produce the key. For `LinkedMap`s & `DoubleMap`s there still needs to be a specific section, there are a number of gotchas.

Reply:
Added these cases to the outstanding . issues in `.TODO.md`